src/Resources/ConfigResource.php

```diff
@@ -1,52 +1,82 @@
 <?php
 namespace GravApi\Resources;
 
+use GravApi\Config\Config;
+use GravApi\Config\Constants;
+use GravApi\Models\ConfigModel;
+use GravApi\Resources\Resource;
+
 /**
  * Class ConfigResource
  * @package GravApi\Resources
  */
-class ConfigResource
+class ConfigResource extends Resource
 {
-    protected $config;
     protected $id;
 
-    public function __construct($config)
+    /**
+     * @param ConfigModel $config
+     * @return ConfigResource
+     */
+    public function __construct(ConfigModel $config)
     {
-        // we should only ever have a single item array
-        foreach ($config as $key => $value) {
-            $this->id = $key;
-        }
+        $this->id = $config->id;
+        $this->resource = $config->data;
```

Comment:
Maybe a better name for this would be `attributes` or `data`, instead of `resource`?